addons/html_editor/static/src/main/media/icon_plugin.js

```diff
@@ -0,0 +1,115 @@
+    static dependencies = ["history", "link", "selection"];
+
+    static resources(p) {
+        return {
+            toolbarNamespace: [
+                {
+                    id: "icon",
+                    isApplied: (traversedNodes) => {
+                        return traversedNodes?.[0]?.classList?.contains("fa");
+                    },
+                },
+            ],
+            toolbarGroup: [
+                {
+                    id: "icon_zoom",
+                    sequence: 1,
+                    namespace: "icon",
+                    buttons: [
+                        {
+                            id: "icon_zoom_1",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "1");
+                            },
+                            text: "1x",
+                            name: _t("Icon size 1x"),
+                            isFormatApplied: () => p.hasIconZoom("1"),
+                        },
+                        {
+                            id: "icon_zoom_2",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "2");
+                            },
+                            text: "2x",
+                            name: _t("Icon size 2x"),
+                            isFormatApplied: () => p.hasIconZoom("2"),
+                        },
+                        {
+                            id: "icon_zoom_3",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "3");
+                            },
+                            text: "3x",
+                            name: _t("Icon size 3x"),
+                            isFormatApplied: () => p.hasIconZoom("3"),
+                        },
+                        {
+                            id: "icon_zoom_4",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "4");
+                            },
+                            text: "4x",
+                            name: _t("Icon size 4x"),
+                            isFormatApplied: () => p.hasIconZoom("4"),
+                        },
+                        {
+                            id: "icon_zoom_5",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "5");
+                            },
+                            text: "5x",
+                            name: _t("Icon size 5x"),
+                            isFormatApplied: () => p.hasIconZoom("5"),
+                        },
+                    ],
+                },
+            ],
+        };
+    }
+
+    getSelectedIcon() {
+        const selectedNodes = this.shared.getSelectedNodes();
+        return selectedNodes.find((node) => node.classList?.contains?.("fa"));
+    }
+
+    handleCommand(command, payload) {
+        switch (command) {
+            case "icon_zoom":
```

Comment:
I though we were always using upper case?

Why do you call it zoom (here and in function names, etc) if its about size / resize ?